### PR TITLE
Sketchbook

### PR DESCRIPTION
**Pull request by matsarat**

### CODE REVIEW

**Review by Czeffik — APPROVED** [review]

**Comment by Czeffik** [comment]
Merge and remove `sketchbook` branch

File: A_arithmetic_progression.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from data_reader import DataReader
+data_reader = DataReader("sums_in_loop.txt")
+
+data = data_reader.get_table_int(" ")
+print(data)
+
+#first value opens the sequence
```
Comment (Czeffik) [inline A_arithmetic_progression.py:7]: FYI - we should not commit commented code. If it is not necessary to run your program - simple remove it

File: Tic_tac_toe.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+
+def get_validated_coordinates(player, board, number_of_rows, number_of_columns):
+    print("Now plays: " + player)
+    input_row = input("Please insert row number: ")
+    input_column = input("Please insert column number: ")
+    if input_row == "" or input_column == "":
+        print("Please insert valid coordinates!")
+        return get_validated_coordinates(player, board, number_of_rows, number_of_columns)
+    move_row = (int(input_row) - 1)
```
Comment (Czeffik) [inline Tic_tac_toe.py:24]: hmm... didn't you have to add validation, that `input_row` and `input_column` is greater than 0?

File: Tic_tac_toe.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        print("The field is already taken!")
+        return get_validated_coordinates(player, board, number_of_rows, number_of_columns)
+    return coordinates
+
+
+def mark_board_with_player_move(coordinates, player, board):
+    board[coordinates[0]][coordinates[1]] = player
+
+
+def is_current_player_winner(array, number_of_winning_marks, current_player):
```
Comment (Czeffik) [inline Tic_tac_toe.py:40]: shouldn't you return `False` somewhere in this method?

**Comment by Czeffik** [comment]
When you are creating new PR add me as reviewer so I will have a notification that I should look on something ;)

image

File: Tic_tac_toe.py (hunk 3 above)
Comment (matsarat) [reply Tic_tac_toe.py:40]: Done.

File: Tic_tac_toe.py (hunk 2 above)
Comment (matsarat) [reply Tic_tac_toe.py:24]: Done.

File: A_arithmetic_progression.py (hunk 1 above)
Comment (matsarat) [reply A_arithmetic_progression.py:7]: Done.

**Pull request merged**